store.go

```diff
@@ -235,6 +236,73 @@ func (store *Store) Delete(ctx context.Context, prefix string, index int64) (ok
 	return
 }
 
+// Session return the session used to lock the given key. It returns an error
+// if the key is not locked or does not exist.
+func (store *Store) Session(ctx context.Context, key string) (sess Session, err error) {
+	var keyData KeyData
+	if keyData, err = store.readKeyData(ctx, key); err != nil {
+		return
+	}
+
+	if len(keyData.Session) == 0 {
+		err = fmt.Errorf("key %s is not locked", key)
+		return
+	}
+
+	var res io.ReadCloser
+	var configs []sessionConfig
+
+	path := "/v1/session/info/" + string(keyData.Session)
+	if _, res, err = store.client().do(ctx, "GET", path, nil, nil); err != nil {
+		return
+	}
+	defer res.Close()
+
+	dec := json.NewDecoder(res)
+	if err = dec.Decode(&configs); err != nil {
+		return
+	}
+
+	if len(configs) == 0 {
+		err = fmt.Errorf("no session data for key %s", key)
+		return
+	}
+
+	config := configs[0]
+	sess = Session{
+		Client:    store.client(),
+		ID:        keyData.Session,
+		Name:      config.Name,
+		Behavior:  SessionBehavior(config.Behavior),
+		LockDelay: fromSeconds(config.LockDelay),
+		TTL:       fromSeconds(config.TTL),
```

Comment:
What do you think about changing the `TTL` field to be of type `time.Duration`, and changing `LockDelay` to be of type `float64`, it seems like it would match the types that consul returns.

Then converting from `float64` to `time.Duration` can be done with something like that I think 
```go
func duration(f float64) time.Duration {
    s := time.Duration(f) * time.Second
    ns := time.Duration(f * float64(time.Second)) % time.Second
    return s + ns
}
```

Reply:
If you are referring to the ```Session``` struct, I would rather not modify something already used by other repo.

For the ```sessionConfig``` struct, it is already used when creating a session https://github.com/segmentio/consul-go/blob/master/session.go#L86.
Also in the sample given in consul documentation:
```json
{
  "LockDelay": "15s",
  "Name": "my-service-lock",
  "Node": "foobar",
  "Checks": ["a", "b", "c"],
  "Behavior": "release",
  "TTL": "30s"
}
```
The ```LockDelay``` and ```TTL``` are given as string. I'm worry about modifying something which is already working.

Reply:
Fair, I was a bit concerned about the hacky `fromSeconds` function which seems to assume there may be multiple formats on those fields.